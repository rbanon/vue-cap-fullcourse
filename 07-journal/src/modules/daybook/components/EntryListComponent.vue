<template>
  <div ckass="entry-list-container">
    <div class="px-2 pt-2">
      <input type="text" class="form-control" placeholder="Buscar entradas" />
    </div>
    <div class="entry-scrollarea px-2 pt-2">
      <EntryComponent v-for="item in 100" :key="item" />
    </div>
  </div>
</template>

<script>
import { defineAsyncComponent } from "vue";
export default {
  name: "EntryListComponent",
  components: {
    EntryComponent: defineAsyncComponent(() => import("./EntryComponent.vue")),
  },
};
</script>

<style lang="scss" scoped>
.entry-list-container {
  background-color: #2c2c2c;
  height: calc(100vh - 56px);
}

.entry-scrollarea {
  height: calc(100vh - 104px);
  overflow-y: auto;
}
</style>
