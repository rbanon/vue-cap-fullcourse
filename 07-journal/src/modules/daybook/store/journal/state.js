export default () => ({
  isLoading: true,
  entries: [
    {
      id: new Date().getTime(),
      date: new Date().toDateString(),
      text: "Ipsum ullamco proident tempor dolor proident nisi esse adipisicing labore irure mollit sunt laboris qui. Aliqua laboris nulla cillum in laborum do sunt nisi. Est aliquip cupidatat eiusmod tempor sunt reprehenderit cillum ullamco. Velit tempor aute ea ullamco aute aliquip excepteur laborum consectetur.",
      picture: null,
    },
    {
      id: new Date().getTime() + 1000,
      date: new Date().toDateString(),
      text: "Nulla culpa elit est enim nulla aliqua occaecat tempor. Ea in nisi do adipisicing irure voluptate occaecat esse Lorem aliquip ipsum. Pariatur consequat occaecat excepteur aliqua Lorem anim. Amet minim velit mollit aliquip irure ut pariatur pariatur ipsum. Ad tempor ex eu duis culpa adipisicing do aliquip eiusmod sint laboris velit excepteur. Amet proident sit et enim sunt non duis non dolore dolore reprehenderit sit commodo.",
      picture: null,
    },
    {
      id: new Date().getTime() + 2000,
      date: new Date().toDateString(),
      text: "Pariatur amet voluptate occaecat voluptate veniam. Reprehenderit exercitation laboris aute ex reprehenderit commodo qui ullamco pariatur eu. Esse proident deserunt incididunt quis consequat cillum laborum. Cupidatat duis sunt irure laborum cillum voluptate exercitation in sint sint consectetur aute. Reprehenderit laborum irure aliquip sint non fugiat deserunt fugiat.",
      picture: null,
    },
  ],
});
