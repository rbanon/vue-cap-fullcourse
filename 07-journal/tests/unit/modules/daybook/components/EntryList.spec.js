import { createStore } from "vuex";
import { VueWrapper, shallowMount } from "@vue/test-utils";
import EntryListComponent from "@/modules/daybook/components/EntryListComponent.vue";
import { journalState } from "../../../mock-data/test-journal-state";
import journal from "@/modules/daybook/store/journal";

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  });
describe("Pruebas en el Entry List", () => {
  // const journalMockModule = {
  //   namespaced: true,
  //   getters: { getEntriesByTerm },
  //   state: () => ({
  //     isLoading: false,
  //     entries: journalState.entries,
  //   }),
  // };

  // const mockRouter = {
  //   push: jest.fn(),
  // };

  // const store = createStore({
  //   modules: {
  //     journal: {
  //       ...journalMockModule,
  //     },
  //   },
  // });

  const store = createVuexStore(journalState);

  const mockRouter = {
    push: jest.fn(),
  };

  let wrapper = VueWrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryListComponent, {
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });
  it("Hace match con el snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Debe de llamar el getEntriesByTerm sin termino y mostrar 2 entradas", () => {
    console.log(wrapper.html());

    expect(wrapper.findAll("entry-component-stub").length).toBe(2);
  });

  it("Debe de llamar el getEntriesByTerm y filtrar las entradas", async () => {
    const input = wrapper.find("input");
    await input.setValue("Probando");

    expect(wrapper.findAll("entry-component-stub").length).toBe(1);
  });

  it("El boton de nuevo debe de redireccionar a /new", () => {
    wrapper.find("button").trigger("click");

    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: { id: "new" },
    });
  });
});
