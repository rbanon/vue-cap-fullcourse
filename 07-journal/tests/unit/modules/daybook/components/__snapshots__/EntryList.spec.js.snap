// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry List Hace match con el snapshot 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar entradas"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> Nueva entrada </button></div>
  <div class="entry-scrollarea px-2 pt-2">
    <entry-component-stub entry="[object Object]"></entry-component-stub>
    <entry-component-stub entry="[object Object]"></entry-component-stub>
  </div>
</div>
`;
